sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/implementation/models/SearchFirstPageResponseWrapper.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.implementation.models;
+
+import com.azure.search.documents.util.SearchPagedResponse;
+
+/**
+ * This class is a wrapper on first page response from search result.
+ */
+public class SearchFirstPageResponseWrapper {
```

Comment:
I don't think this wrapper is necessary. You could directly use the `SearchPagedResponse`.

Reply:
This is what I use to expose the metadata between each paging call. Also the class serves as local caching.

We cannot expose metadata API in SearchPageResponse, so I put the wrapper under implementation.

Reply:
Talked offline. There is no hurt to have a wrapper here. Merge the PR for now. Will try to see if there is reactor api which can cache and reuse the publisher.